lib/Rest/HttpRequest.cpp

```diff
@@ -720,13 +735,22 @@ void HttpRequest::setHeader(char const* key, size_t keyLength,
     // This can be much more elaborated as the can specify weights on encodings
     // However, for now just toggle on deflate if deflate is requested
     _acceptEncoding = EncodingType::DEFLATE;
-  } else if (keyLength == StaticStrings::ContentTypeHeader.size() &&
-             valueLength == StaticStrings::MimeTypeVPack.size() &&
-             memcmp(key, StaticStrings::ContentTypeHeader.c_str(), keyLength) == 0 &&
-             memcmp(value, StaticStrings::MimeTypeVPack.c_str(), valueLength) == 0) {
-    _contentType = ContentType::VPACK;
-    // don't insert this header!!
-    return;
+  } else if ((_contentType == ContentType::UNSET) &&
+             (keyLength == StaticStrings::ContentTypeHeader.size()) &&
+             (memcmp(key, StaticStrings::ContentTypeHeader.c_str(), keyLength) == 0)) {
+    if (valueLength == StaticStrings::MimeTypeVPack.size() &&
+        memcmp(value, StaticStrings::MimeTypeVPack.c_str(), valueLength) == 0) {
+      _contentType = ContentType::VPACK;
+      // don't insert this header!!
+      return;
+    }
+    if (valueLength >= StaticStrings::MimeTypeJsonNoEncoding.size() &&
+        memcmp(value, StaticStrings::MimeTypeJsonNoEncoding.c_str(), valueLength) == 0) {
+      _contentType = ContentType::JSON;
+      // don't insert this header!!
+      return;
+    }
+    _contentType = ContentType::UNSET;
```

Comment:
this line is superfluous, as we can only get here if `_contentType` is `ContentType::UNSET` already.